wit/mqtt.wit

```diff
@@ -0,0 +1,32 @@
+interface mqtt {
+  /// Errors related to interacting with Mqtt
+  variant error {
+      /// An invalid address string
+      invalid-address,
+      /// There are too many open connections
+      too-many-connections,
+      /// A retrieved value was not of the correct type
+      type-error,
+      /// Some other error occurred
+      other(string),
+  }
+
+  /// QoS for publishing Mqtt messages
+  /// TODO: will enum result in consistent values across languages?
```

Comment:
cc @itowlson @dicej @rylev

Reply:
Sort of. There is an index associated with each enum variant but it isn't necessarily exposed to guest languages.